Clean up omiserver.conf

JumpingYang001 opened the pull request:
@Microsoft/omi-devs , could you help to review this PR? thanks very much!
here are changes:
1. Remove trace setting for both Unix/installbuilder/conf/omiserver.conf and Unix/etc/omiserver.conf (trace setting doesn't work, so remove it.)
2. Update loglevel section for Unix/etc/omiserver.conf
3. Add loglevel section for Unix/installbuilder/conf/omiserver.conf 
4. Remove nonroot setting for both Unix/installbuilder/conf/omiserver.conf and Unix/etc/omiserver.conf
5. Update NICKNAME section description

For NICKNAME section, I have tried all below setting, they works fine:
```
prefix=GNU
libdir=/opt/omi/lib
bindir=/opt/omi/bin
localstatedir=/var/opt/omi
sysconfdir=/etc/opt/omi/conf
providerdir=/etc/opt/omi/conf
certsdir=/etc/opt/omi/ssl
datadir=/opt/omi/share
rundir=/var/opt/omi/run
logdir=/var/opt/omi/log
schemadir=/opt/omi/share/omischema
schemafile=/opt/omi/share/omischema/CIM_Schema.mof
pidfile=/var/opt/omi/run/omiserver.pid
logfile=/var/opt/omi/log/omiserver.log
registerdir=/etc/opt/omi/conf/omiregister
pemfile=/etc/opt/omi/ssl/omi.pem
keyfile=/etc/opt/omi/ssl/omikey.pem
agentprogram=/opt/omi/bin/omiagent
serverprogram=/opt/omi/bin/omiserver
includedir=/opt/omi/include
configfile=/etc/opt/omi/conf/omiserver.conf
```

jeffaco commented on Unix/installbuilder/conf/omiserver.conf:
```diff
@@ -16,9 +16,11 @@ httpsport=0
+##   If commented out, then default value is: WARNING
 ##
-#trace=(true|false)
+#loglevel = INFO
```
Generally, when we comment out, uncommenting gets you the default. See, for example, `/etc/ssh/sshd_config`. Thus, suggest changing commented out line above to WARNING.

jeffaco commented on Unix/installbuilder/conf/omiserver.conf:
```diff
@@ -33,7 +35,7 @@ httpsport=0
 #NtlmCredsFile=/etc/opt/omi/.creds/ntlm
```
Is this the default? Or is this disabled if commented out? Comments indicate that you must comment out the location of the credentials file, just wondering if this is actually the case (in real life) or not. Check with @yakman2020 if unsure.

JumpingYang001 replied:
in fact, it is disabled ntlm authentication if commented out.

jeffaco commented on Unix/installbuilder/conf/omiserver.conf:
```diff
@@ -33,7 +35,7 @@ httpsport=0
 
 ##
-## <NICKNAME> -- set the value of nickname.
+## <NICKNAME> -- set the value of nickname for omi related directory and file path.
```
What, exactly, is a "NICKNAME"? What does it do? Here I am, working with OMI for a long time, and even I don't know this. Clarify in this comment please.

JumpingYang001 replied:
here is a customized configuration for omi directories and files, like omiserver or omiagent path, omi log directory and path, ssl directory and ssl keypem path, etc.

jeffaco commented on Unix/installbuilder/conf/omiserver.conf:
```diff
@@ -33,7 +35,7 @@ httpsport=0
-## <NICKNAME> -- set the value of nickname.
+## <NICKNAME> -- set the value of nickname for omi related directory and file path.
 ##
 #prefix=PATH
```
Do all these things actually work if overridden? Did you try changing to alternate paths to verify this?

I'm asking this for lines 38-45, 47-49, 55-57. I believe the non-commented out lines, 46, 50-56, and 58 are actually required because OMI defaults improperly for what installbuilder sets up.

Paul would like this to be "cleaned up". What this means, in order of priority:

1. If these are here, we need to verify they work, not just with default location, but if moved to a new location, does everything still work properly?
2. We need to decide if we want to support moving all of these things?
3. For things that we depend on (like the un-commented out lines, if we indeed still depend on them), they should probably be moved to a different section with a note to not change or system may not work properly.

This is going to take some work to figure out on a per-line basis. If you want to take this up, awesome. But do all the work that we need so we don't need to revisit this and can close out the related issue completely.

JumpingYang001 replied:
will answer the 3 questions later today, thanks

JumpingYang001 replied:
1. None of the directory path moved to new location will work, I have tried one by one to move to /tmp. OMI related binaries now are very depended, it doesn't work now! 
I have got following errors when move things to /tmp:
error 1: 
```
/tmp/opt/omi/bin/omiengine: Protocol_New_Listener() failed: /var/opt/omi/run/omiserver.sock2017/12/06 20:43:05 [28658,28658] ERROR: null(0): Protocol_New_Listener() failed:
```
error 2:
```
/tmp/opt/omi/bin/omiserver: Launch failed: 22017/12/06 20:43:43 [28667,28667] ERROR: null(0): Launch failed: 409012848
```
error 3:
```
2017/12/06 20:57:28 [30984,30984] FATAL: null(0): EventId=10003 Priority=FATAL failed to initialize the dispatcher: 1
/opt/omi/bin/omiserver: failed to initialize the dispatcher: 12017/12/06 20:57:28 [30984,30984] ERROR: null(0): failed to initialize the dispatcher: 768683504
```
2. We don't support moving omi things anymore now.
3. Yes, they should not be changed and should notice that in a section, I will update them, thanks.

JumpingYang001 replied:
Forget to say that I tried to moving with files under the 3 directory and also individual file, but none of them work.
/tmp/opt
/tmp/var/opt
/tmp/etc/opt/omi
```
prefix=GNU
libdir=/tmp/opt/omi/lib
includedir=/tmp/opt/omi/include

bindir=/tmp/opt/omi/bin
agentprogram=/tmp/opt/omi/bin/omiagent
serverprogram=/tmp/opt/omi/bin/omiserver

datadir=/tmp/opt/omi/share
schemadir=/tmp/opt/omi/share/omischema
schemafile=/tmp/opt/omi/share/omischema/CIM_Schema.mof

localstatedir=/tmp/var/opt/omi

rundir=/tmp/var/opt/omi/run
pidfile=/tmp/var/opt/omi/run/omiserver.pid

logdir=/tmp/var/opt/omi/log
logfile=/tmp/var/opt/omi/log/omiserver.log

certsdir=/tmp/etc/opt/omi/ssl
pemfile=/tmp/etc/opt/omi/ssl/omi.pem
keyfile=/tmp/etc/opt/omi/ssl/omikey.pem

sysconfdir=/tmp/etc/opt/omi/conf
providerdir=/tmp/etc/opt/omi/conf
registerdir=/tmp/etc/opt/omi/conf/omiregister
configfile=/tmp/etc/opt/omi/conf/omiserver.conf
```

jeffaco commented on Unix/installbuilder/conf/omiserver.conf:
```diff
@@ -58,6 +60,3 @@ keyfile=/etc/opt/omi/ssl/omikey.pem
 configfile=/etc/opt/omi/conf/omiserver.conf
 #NoSSLv2=true
```
61-62 work, I know because I added them. But they should be in a separate security section or something explaining what they do. I think it's clear from the name, but a line or two of explanation might be good. In general, we disable SSLv2 because it's not considered secure.

jeffaco commented on Unix/installbuilder/conf/omiserver.conf:
```diff
@@ -58,6 +60,3 @@ keyfile=/etc/opt/omi/ssl/omikey.pem
 #NoSSLv3=false
-
-## non-root feature parameters
-#nonroot=true
```
Hip hip hooray. I never liked this here. Even if it worked (and I don't think it does anymore), we don't suggest it and wouldn't support it anymore.

JumpingYang001 commented on Unix/etc/omiserver.conf:
```diff
@@ -16,51 +16,36 @@
 ## if USER is specified within <> as the service account, 
 ##      it will be replaced with $USER when copied to output*
-#nonroot=true
 service=<USER>
```
This USER is still used for building, I don't know if we can remove it, @palladia , how do think this part?

JumpingYang001 commented:
@jeffaco , I have updated based on your comments, could you help to check it again? thanks very much!

jeffaco commented on Unix/etc/omiserver.conf:
```diff
@@ -16,51 +16,36 @@
+##   Valid options are: ERROR, WARNING, INFO, DEBUG, VERBOSE (debug builds only)
+##   If commented out, then default value is: WARNING
 ##
 loglevel = WARNING
```
Comment this out please. We don't want to force the default, we want the default to be commented out.

jeffaco commented on Unix/etc/omiserver.conf:
```diff
@@ -16,51 +16,36 @@
 ##
-## <NICKNAME> -- set the value of nickname.
+## This section is for security protocol settings
+##   Including NoSSLv2,NoSSLv3,NoTLSv1_0,NoTLSv1_1,NoTLSv1_2 and NoSSLCompression.
```
Delete line 27, we'll see what it includes by looking on.

jeffaco commented on Unix/etc/omiserver.conf:
```diff
@@ -16,51 +16,36 @@
-#configfile=PATH
 NoSSLv2=true
 NoSSLv3=false
 NoTLSv1_0=false
```
For lines 31-34, I think these have been dragged along forever. I don't think we give guidance to change these. If we don't, we may want to pull them out. Or, if you leave them, then at least include comments on what, EXACTLY, each one does.

Did you verify that each of these actually work properly? I know that 29-30 did (I tested when I added them). But for 31-34, I have no idea if these work or not.

We need to consider with these:

1. Do we want to leave them in here (suggesting that users can adjust them)?
2. If yes, then we must document what these do, exactly, and we must make sure each of them actually work!

We definitely should include NoSSLv2 and NoSSLv3, although they too should be commented out (verify that the current values are actually default values), and we should include a brief description of what they mean. It think it's obvious based on the name, but then I worked on it, I'm not at all sure that customers would understand that.

JumpingYang001 replied:
will update them later today, thanks.

JumpingYang001 commented:
@jeffaco , I removed the PATHs in installer conf, so you didn't see my comments in new review, I have made them back now. (I guess you were seeing from review page: https://github.com/Microsoft/omi/pull/469/files, in fact, you can see the many "Show outdated" items in PR page: https://github.com/Microsoft/omi/pull/469, all old review comments are there.)
I have commentted all PATH in installer conf, they are default values and works fine after commentted out.
The only PATH can be modify to new location is "logfile", I have updated it in section comment.
For "NoTLS" related setting, I have removed them, thanks.

Previous 3 questions:
1. If these are here, we need to verify they work, not just with default location, but if moved to a new location, does everything still work properly?
Jumping: None of the directory or file path moved to new location will work except "logfile", I have tried one by one to move to /tmp. OMI related binaries now are very depended, it doesn't work now! 

I have got following errors when move things to /tmp:
/tmp/opt
/tmp/var/opt
/tmp/etc/opt/omi

error 1: 
```
/tmp/opt/omi/bin/omiengine: Protocol_New_Listener() failed: /var/opt/omi/run/omiserver.sock2017/12/06 20:43:05 [28658,28658] ERROR: null(0): Protocol_New_Listener() failed:
```
error 2:
```
/tmp/opt/omi/bin/omiserver: Launch failed: 22017/12/06 20:43:43 [28667,28667] ERROR: null(0): Launch failed: 409012848
```
error 3:
```
2017/12/06 20:57:28 [30984,30984] FATAL: null(0): EventId=10003 Priority=FATAL failed to initialize the dispatcher: 1
/opt/omi/bin/omiserver: failed to initialize the dispatcher: 12017/12/06 20:57:28 [30984,30984] ERROR: null(0): failed to initialize the dispatcher: 768683504
```
2.We need to decide if we want to support moving all of these things?
Jumping: We don't support moving omi things anymore now except "logfile".

3.For things that we depend on (like the un-commented out lines, if we indeed still depend on them), they should probably be moved to a different section with a note to not change or system may not work properly.
Jumping: Yes, they should not be changed and should notice that in a section, I have updated them, thanks.

I tried to moving with files under the 5 group directory and also individual file, but none of them work.
```
Group1:
prefix=GNU
libdir=/tmp/opt/omi/lib
includedir=/tmp/opt/omi/include

Group2:
bindir=/tmp/opt/omi/bin
agentprogram=/tmp/opt/omi/bin/omiagent
serverprogram=/tmp/opt/omi/bin/omiserver

Group3:
datadir=/tmp/opt/omi/share
schemadir=/tmp/opt/omi/share/omischema
schemafile=/tmp/opt/omi/share/omischema/CIM_Schema.mof

Group4:
localstatedir=/tmp/var/opt/omi

rundir=/tmp/var/opt/omi/run
pidfile=/tmp/var/opt/omi/run/omiserver.pid

logdir=/tmp/var/opt/omi/log
logfile=/tmp/var/opt/omi/log/omiserver.log

certsdir=/tmp/etc/opt/omi/ssl
pemfile=/tmp/etc/opt/omi/ssl/omi.pem
keyfile=/tmp/etc/opt/omi/ssl/omikey.pem

Group5:
sysconfdir=/tmp/etc/opt/omi/conf
providerdir=/tmp/etc/opt/omi/conf
registerdir=/tmp/etc/opt/omi/conf/omiregister
configfile=/tmp/etc/opt/omi/conf/omiserver.conf
```

JumpingYang001 commented:
From https://github.com/Microsoft/omi/blob/b13e734a8b8cd9c29c2a0ade6cdae57fc2a23299/Unix/server/servercommon.c#L859, it shows the current NoSSLv2 and NoSSLv3 values are equal to default commented the 2 settings.
I will comment these 2 settings out in next change.

JumpingYang001 commented:
@jeffaco , I have updated based on your comments, could you help to check it again? thanks very much!

I have removed 4 settings in installer conf file since they doesn't effect anything:
```
certsdir=/tmpcert # this setting doesn't effect anything.
localstatedir=/tmplocal # this setting doesn't effect anything.
includedir==/tmp6 # this is only for building, we can remove it from installer conf file.
libdir=/tmp7 # this is only for building, we can remove it from installer conf file.
```

jeffaco commented on Unix/installbuilder/conf/omiserver.conf:
```diff
@@ -33,31 +35,49 @@ httpsport=0
-#serverprogram=PATH
-#includedir=PATH
-configfile=/etc/opt/omi/conf/omiserver.conf
+## This is the setting for omi related directory and file path, the following values are default values, and we don't support modifing the path.
```
Limit lines to 80 bytes or less in configuration files please.

jeffaco replied:
Run-on sentence.

JumpingYang001 replied:
Removed commented default values section.

jeffaco commented on Unix/installbuilder/conf/omiserver.conf:
```diff
@@ -33,31 +35,49 @@ httpsport=0
-configfile=/etc/opt/omi/conf/omiserver.conf
+## This is the setting for omi related directory and file path, the following values are default values, and we don't support modifing the path.
+##   Please don't modify any path value, otherwise the omi will not work!
+##
```
How about changing 38-39 to something like:

```
## Settings for omi-related directory and file paths.
##   Do not change any values below. Otherwise, OMI won't function properly!
```

jeffaco replied:
But wait: If these are all DEFAULT values, can't you just get rid of them entirely?

If everything is commented out (default values), and we can't change them due to how we build source code, can't the entire segment be removed from this file?

Please validate and resubmit your review, thanks.

JumpingYang001 replied:
Yes, we can remove the default values, I will update them later, thanks.

JumpingYang001 replied:
Removed commented default values section.

JumpingYang001 commented:
After investigation for sysconfdir, even it can be modified as other location, but that doesn't make sense, as the omiregister folder and omiserver.conf can't be modified as following, so I think we should keep it as default value, so just remove it in installer conf file.
```
sysconfdir=/etc/opt/omi/conf
providerdir=/etc/opt/omi/conf
registerdir=/etc/opt/omi/conf/omiregister
configfile=/etc/opt/omi/conf/omiserver.conf
```

JumpingYang001 commented:
After modifying /etc/opt/omi/conf/omiserver.conf setting to
```
logdir=/tmp
logfile=/tmp/omiserver.log
```
, it shows following directory and files:
```
~
root@omi64-cent7-01  # ll /tmp
total 20
drwxrwxrwt.  7 root root 4096 Dec 11 21:57 .
drwxr-xr-x. 17 root root 4096 Dec 11 13:31 ..
drwxrwxrwt.  2 root root    6 Aug  6  2014 .font-unix
drwxrwxrwt.  2 root root    6 Aug  6  2014 .ICE-unix
-rw-r--r--.  1 root root    0 Dec 11 21:54 omiagent.root.root.log
-rw-r--r--.  1 root root 1577 Dec 11 21:55 omiserver.conf
-rw-r--r--.  1 root root 2021 Dec 11 21:57 omiserver.log
drwxrwxrwt.  2 root root    6 Aug  6  2014 .Test-unix
-rw-------.  1 root root  240 Dec 11 21:36 tmp.Sn9wyM1gHt
drwxrwxrwt.  2 root root    6 Aug  6  2014 .X11-unix
drwxrwxrwt.  2 root root    6 Aug  6  2014 .XIM-unix
You have new mail in /var/spool/mail/root

~
root@omi64-cent7-01  # ll /var/opt/omi/log/
total 28
drwxr-xr-x. 2 omi  omi  4096 Dec 11 21:56 .
drwxr-xr-x. 8 root root   75 Dec 11 21:36 ..
-rw-r--r--. 1 root root   94 Dec 11 21:56 miclient.log
-rw-r--r--. 1 root root 5716 Dec 11 21:56 omiclient-recv.trc
-rw-r--r--. 1 root root 1507 Dec 11 21:56 omiclient-send.trc
-rw-r--r--. 1 root root  143 Dec 11 21:36 omiserver.log
-rw-r--r--. 1 omi  omi  1413 Dec 11 21:56 omiserver-recv.trc

```

Here is what I am seeing for the modification effect:
 - copy /etc/opt/omi/conf/omiserver.conf  to omiserver.conf, I don't know why this happen.
 - Generate omiagent.root.root.log and omiserver.log under /tmp
 - /var/opt/omi/log/miclient.log should be set by default omicli.conf
 - /var/opt/omi/log/*.trc files are trace logs, I don't know why it still use default path instead of my setting /tmp, that's a confuse here.
 - /var/opt/omi/log/omiserver.log is the old log, it is by design when omi started with installing.

JumpingYang001 commented:
@jeffaco , can you help to check my previous comment about the modification effect of "logdir" and "logfile" ? thanks very much for helping!

jeffaco commented on Unix/etc/omiserver.conf:
```diff
@@ -1,4 +1,4 @@
-# omiserver configuration file
+# omiserver configuration file, more detail: https://github.com/Microsoft/omi/#configuring-omi-server
```
This is a `.conf` file, restrict lines to 80 bytes maximum please. Didn't I mention that in last review?

jeffaco commented on Unix/installbuilder/conf/omiserver.conf:
```diff
@@ -1,4 +1,4 @@
-# omiserver configuration file
+# omiserver configuration file, more detail: https://github.com/Microsoft/omi/#configuring-omi-server
```
Ditto. In a `.conf` file, lines should be limited to 80 bytes in length please.

JumpingYang001 commented:
@jeffaco , I have fixed the issues based on your comments, could you help to review it again? thanks very much!
